dlp/src/main/java/dlp/snippets/InspectStringCustomExcludingSubstring.java

```diff
@@ -0,0 +1,120 @@
+      String customDetectorPattern, List<String> excludedSubstringList) {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DlpServiceClient dlp = DlpServiceClient.create()) {
+      // Specify the type and content to be inspected.
+      ByteContentItem byteItem =
+          ByteContentItem.newBuilder()
+              .setType(BytesType.TEXT_UTF8)
+              .setData(ByteString.copyFromUtf8(textToInspect))
+              .build();
+      ContentItem item = ContentItem.newBuilder().setByteItem(byteItem).build();
+
+      // Specify the type of info the inspection will look for.
+      InfoType infoType = InfoType.newBuilder().setName("CUSTOM_NAME_DETECTOR").build();
+      CustomInfoType customInfoType = CustomInfoType.newBuilder()
+          .setInfoType(infoType).setRegex(
+              Regex.newBuilder().setPattern(customDetectorPattern)).build();
+
+      // Exclude partial matches from the specified excludedSubstringList.
+      ExclusionRule exclusionRule = ExclusionRule.newBuilder()
+          .setMatchingType(MatchingType.MATCHING_TYPE_PARTIAL_MATCH)
+          .setDictionary(Dictionary.newBuilder()
+              .setWordList(WordList.newBuilder().addAllWords(excludedSubstringList)))
+          .build();
+
+      // Construct a ruleset that applies the exclusion rule to the EMAIL_ADDRESSES infotype.
+      InspectionRuleSet ruleSet = InspectionRuleSet.newBuilder()
+          .addInfoTypes(infoType)
+          .addRules(InspectionRule.newBuilder().setExclusionRule(exclusionRule))
+          .build();
+
+      // Construct the configuration for the Inspect request, including the ruleset.
+      InspectConfig config =
+          InspectConfig.newBuilder()
+              .addCustomInfoTypes(customInfoType)
+              .setIncludeQuote(true)
+              .addRuleSet(ruleSet)
+              .build();
+
+      // Construct the Inspect request to be sent by the client.
+      InspectContentRequest request =
+          InspectContentRequest.newBuilder()
+              .setParent(LocationName.of(projectId, "global").toString())
+              .setItem(item)
+              .setInspectConfig(config)
+              .build();
+
+      // Use the client to send the API request.
+      InspectContentResponse response = dlp.inspectContent(request);
+
+      // Parse the response and process results
+      System.out.println("Findings: " + response.getResult().getFindingsCount());
+      for (Finding f : response.getResult().getFindingsList()) {
+        System.out.println("\tQuote: " + f.getQuote());
+        System.out.println("\tInfo type: " + f.getInfoType().getName());
+        System.out.println("\tLikelihood: " + f.getLikelihood());
+      }
+    } catch (Exception e) {
```

Comment:
It would be nice if you were more specific here and say what might happen and how they might mitigate it.
https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions

Reply:
This is copied from existing parallel sample code. I agree that it's pretty crappy, and we're also not really consistent: sometimes we do this, sometimes we just stick a "throws IOException" on the function. 

Of the two I think I prefer the latter, it makes it clear that this code is explicitly not handling that. Can I handle these in a  follow-up for all our DLP Java samples? (see internal bug b/158239329)

Reply:
SGTM